app/helpers/duties_helper.rb

```diff
@@ -42,19 +42,45 @@ def process_day_place(duties)
     result
   end
 
-  def process_duty(duties)
-    colspan = 0
+  def check_condition(span_free, span_user_id, span_request_user_id, current_duty)
+    span_free != current_duty.free || (!span_free && !current_duty.free && ((!span_request_user_id.nil? && span_request_user_id != current_duty&.request_user_id) || span_user_id != current_duty&.user&.id))
+  end
+
+  def get_index_array(duties)
+    index_array = [0]
+    span_free = duties[0].free
+    span_user_id = duties[0]&.user&.id
```

Comment:
`&.user_id`

Reply:
Actually probably better to use `.first` instead of `[0]` so that intent is clearer